### ISSUE TITLE
kld101 laser controller -- unpacker fails to read message

### ISSUE THREAD

**Issue by ianzur** [yaq-project/thorlabs-apt-protocol#16]
I started working on a daemon for kld101, but ran into some issues.

error: (but this message is readable)
```fish
> python test_thorlabs-apt-protocol_kld101.py
/home/ianzur/Documents/projects/yaq-project/thorlabs-apt-protocol/thorlabs_apt_protocol/unpacker.py:52: UserWarning: Invalid message with id=0x0802
b'\x02\x08\x04\x00\x81P'
  warnings.warn(message)
```

code:
```py
### 14 Feb. 2023 
# test this thorlabs apt protocol implementation with the laser diode driver

import thorlabs_apt_protocol as apt
import serial

port = serial.Serial("/dev/ttyUSB0", 115200, rtscts=True, timeout=0.1)
port.rts = True
port.reset_input_buffer()
port.reset_output_buffer()
port.rts = False
port.write(apt.hw_no_flash_programming(source=1, dest=0x50))

unpacker = apt.Unpacker(port, "warn")

# request hw info
#port.write(apt.hw_req_info(source=1, dest=0x50))

# enable laser
#port.write(apt.la_enableoutput(source=1, dest=0x50))
#print(apt.la_enableoutput(source=1, dest=0x50))

# set laser power to max (TODO: CONSTANTS) 
#port.write(apt.la_set_power_setpoint(source=1, dest=0x50, setpoint=32767))

# These should return??
port.write(apt.la_req_power_setpoint(source=1, dest=0x50))

#port.write(apt.la_req_laser_current_and_power(source=1, dest=0x50))

for msg in unpacker:
    print(msg)
```

apt manual -- get msgid highlighted (pg. 297):
![apt_man_shot](https://user-images.githubusercontent.com/33916505/219128921-8f2c0574-ecf3-4f71-8351-2d0506559461.png)

apt manual -- expected get for la_req_power_setpoint (pg. 298):
![image](https://user-images.githubusercontent.com/33916505/219130500-18ba746f-e442-4aa9-b0fd-772ab149420f.png)

I think unpacker msgid errors should at least be logged, not silenced. Took me a while to figure out what was going on.

**Comment by ksunden** [comment]
https://github.com/yaq-project/thorlabs-apt-protocol/blob/1bea2ccb6869d2e2a59f47858e4e59735ed97367/thorlabs_apt_protocol/unpacker.py#L30

The `on_error` parameter of the `Unpacker` allows you to toggle this behavior. I'd be open to switching the default to warn. I don't recall any specific reason we silenced by default there.

This is a newer device and I have not done a sit down with the APT manual in a while (as evidenced by #14 which adds support for other new devices)

Unfortunately there is not a specific changelog, so I have to pour over basically the full manual to make sure there are not changes (even to existing messages, which does happen).

**Comment by ianzur** [comment]
I only went through the commands related for just my device and it was mind numbing. 

I'll make the on_error change in a separate PR. 

Bravo for the work you put in!

**Comment by ksunden** [comment]
Closed by #17
